internal/controller/compute/statefulserverset/statefulserverset.go

```diff
@@ -116,9 +129,14 @@ func (e *external) Create(ctx context.Context, mg resource.Managed) (managed.Ext
 	cr.Status.SetConditions(xpv1.Creating())
 
 	e.log.Info("Creating a new ServerSet", "replicas", cr.Spec.ForProvider.Replicas)
-	for i := 0; i < cr.Spec.ForProvider.Replicas; i++ {
-		e.log.Info("Creating a the data volumes")
-
+	for replicaIndex := 0; replicaIndex < cr.Spec.ForProvider.Replicas; replicaIndex++ {
+		e.log.Info("Creating the data volumes")
+		for volumeIndex := range cr.Spec.ForProvider.Volumes {
+			err := e.dataVolumeController.Ensure(ctx, cr, replicaIndex, volumeIndex)
+			if err != nil {
+				return managed.ExternalCreation{}, err
+			}
+		}
```

Comment:
We could extract this as a method:
```go
err := e.createDataVolumes(ctx, cr, replicaIndex)
if err != nil {
	return managed.ExternalCreation{}, err
}

...
func (e *external) createDataVolumes(ctx context.Context, cr *v1alpha1.StatefulServerSet, replicaIndex int) error {
	e.log.Info("Creating the data volumes")
	for volumeIndex := range cr.Spec.ForProvider.Volumes {
		err := e.dataVolumeController.Ensure(ctx, cr, replicaIndex, volumeIndex)
		if err != nil {
			return err
		}
	}
	return nil
}
```

Reply:
will keep the name with ensure, just to make it clear that it's creating datavolumes only if they don't exist